eppo_core/src/events/event_dispatcher.rs

```diff
@@ -0,0 +1,177 @@
+use crate::events::batch_event_queue::BatchEventQueue;
+use crate::events::event::{Event, GenericEvent};
+use log::info;
+use serde::Serialize;
+use serde_json::to_value;
+use tokio::sync::mpsc::{unbounded_channel, UnboundedReceiver, UnboundedSender};
+use tokio::time::{Duration, Instant};
+
+enum EventDispatcherCommand {
+    Event(GenericEvent),
+    Flush,
+}
+
+#[derive(Debug, Clone)]
+pub struct EventDispatcherConfig {
+    pub sdk_key: String,
+    pub ingestion_url: String,
+    pub delivery_interval: Duration,
+    pub retry_interval: Duration,
+    pub max_retry_delay: Duration,
+    pub max_retries: Option<u32>,
+}
+
+pub struct EventDispatcher {
+    config: EventDispatcherConfig,
+    batch_queue: BatchEventQueue,
+    tx: UnboundedSender<EventDispatcherCommand>,
+    rx: UnboundedReceiver<EventDispatcherCommand>,
```

Comment:
even though `rx` is unused, I had to keep it in scope here to avoid it being dropped. if `rx` is dropped, `tx.send` fails
<img width="648" alt="image" src="https://github.com/user-attachments/assets/3ff6061b-b105-4bcb-b19a-fd08bcb77a75" />

Reply:
that makes sense, thank you. Will update